tools/upload_text.php

```diff
@@ -147,27 +150,33 @@
     $extras = array();
     $back_url = "$code_url/project.php?id=$projectid&amp;expected_state=$new_state";
     $back_blurb = _("Project Page");
-    $deadline = time() + ($weeks * 60 * 60 * 24 * 7);
-
 }
 else if(!$stage)
 {
     // this may be due to a timeout when uploading big files.
     include_once($relPath.'slim_header.inc');
-    
+
     slim_header(_("Upload failed"));
-    
+
     echo "<p>" . _("The upload failed.") . "</p>\n";
     echo "<p>$big_upload_blurb</p>";
-    
-    echo "<p>" . sprintf(_("Please go <a href='%s'>back</a> and try uploading 
-        the original again or uploading a smaller placeholder instead."), 
+
+    echo "<p>" . sprintf(_("Please go <a href='%s'>back</a> and try uploading
+        the original again or uploading a smaller placeholder instead."),
         "javascript:history.back()") . "</p>";
-    
+
     exit;
 }
 
-if (!isset($action))
+if($extend && !$project->is_available_for_smoothreading())
+{
+    // this can happen if project page was stale
+    echo "<p>" , _("The Smooth Reading deadline for this project has passed and cannot be extended in this way."),
```

Comment:
Should we make this a `class='warning'` too? I could go either way.

Reply:
Yes, I've add class='warning'.

Reply:
But it doesn't work because there is no header with css files.